### ISSUE TITLE
Global shims in projects not managed by rye

### ISSUE THREAD

**Issue by bayashi-cl** [astral-sh/rye#1121]
### Steps to Reproduce

Start Global Shims within the sampleproject.
```sh
rye config --set-bool behavior.global-python=true
git clone https://github.com/pypa/sampleproject.git
cd sampleproject
python +3.11
```

### Expected Result

The Python shell will start.

### Actual Result

The command will fail with error.
```
error: Explicit Python selection is not possible within Rye managed projects.
```

### Version Info

```
❯ rye --version
rye 0.34.0
commit: 0.34.0 (d31340178 2024-05-20)
platform: linux (x86_64)
self-python: cpython@3.12.2
symlink support: true
uv enabled: true
```

**Comment by lvignoli** [comment]
For completeness, simply running `python` gives
```
error: Target Python binary 'python' not found in project. Most likely running 'rye sync' will resolve this.
```

What was the reasoning behind this default behaviour?

**Comment by kivo360** [comment]
I'm gonna add to this. The global behavior isn't very good. I have to do lots of work arounds to make it work.

**Comment by lvignoli** [comment]
It seems @bayashi-cl has it right with #1122: for python projects not managed by rye (so no `tool.rye.managed = true`), rye should probably use global python all the time, through global shims if activated.

**Comment by mitsuhiko** [comment]
I agree that the current behavior is not ideal but I think rather than patching around on the behavior it would be better to take a step back and rethink what the actual intended behavior should be for all cases.
